src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/configure-control-plane-monitoring.mdx

```diff
@@ -0,0 +1,499 @@
+
+Additionally, each `endpoint` has:
+* `url`: URL to target, including scheme. Can be `http` or `https`.
+* `insecureSkipVerify`: If set to true, certificate will not be checked for `https` URLs.
+* `auth.type`: Which mechanism to use to authenticate the request. Currently, the following methods are supported:
+ * None: If `auth` is not specified, the request will not contain any authentication whatsoever.
+ * `bearer`: The same bearer token used to authenticate against the Kubernetes API will be sent to this request.
+ * `mtls`: mTLS will be used to perform the request.
+
+##### mTLS [#mtls]
+
+For the `mtls` type, the following needs to be specified:
+
+```yaml
+endpoints:
+  - url: https://localhost:4001
+    auth:
+      type: mtls
+      mtls:
+        secretName: secret-name
+        secretNamespace: secret-namespace
+```
+
+Where `secret-name` is the name of a Kubernetes TLS Secret, which lives in `secret-namespace`, and contains the certificate, key, and CA required to connect to that particular endpoint.
+
+The integration fetches this secret in runtime rather than mounting it, which means it requires an RBAC role granting it access to it. Our Helm Chart automatically detects `auth.mtls` entries at render time and will automatically create entries for these particular secrets and namespaces for you, unless `rbac.create` is set to false.
+
+Our integration accepts a secret with the following keys:
+- `cert`: The PEM-encoded certificate that will be presented to etcd
+- `key`: The PEM-encoded private key corresponding to the certificate above
+
+These certificates should be signed by the same CA etcd is using to operate.
+
+How to generate these certificates is out of the scope of this documentation, as it will vary greatly between different Kubernetes distribution. Please refer to your distribution's documentation to see how to fetch the required etcd peer certificates. In Kubeadm, for example, they can be found in `/etc/kubernetes/pki/etcd/peer.{crt,key}` in the master node.
+
+[//]: # (TODO: This should be a TLS secret when https://github.com/newrelic/nri-kubernetes/issues/342 is fixed.)
+
+Once you have located or generated the etcd peer certificates, you should rename the files to match the keys we expect to be present in the secret, and create the secret in the cluster
+
+```shell
+mv peer.crt cert
+mv peer.key key
+mv ca.crt cacert
+
+kubectl -n newrelic create secret generic newrelic-etcd-tls-secret --from-file=./cert --from-file=./key --from-file=./cacert
+```
+
+Finally, you can input the secret name (`newrelic-etcd-tls-secret`) and namespace (`newrelic`) in the config snippet shown at the beginning of this section. Remember that the Helm Chart will automatically parse this config and create an RBAC role to grant access to this specific secret and namespace for the `nrk8s-controlplane` component, so there's no manual action needed in that regard.
+
+### Static endpoints [#static-endpoints]
```

Comment:
we could also add a small callout telling them they can configure a static endpoint to scrape the KubernetesAPIServer in AWS/GCP with kind=Deployment

Reply:
I'll provide it

Reply:
# if you are wondering why false for nodeAffinity instead of null or {} check https://github.com/helm/helm/issues/9136

```yaml 
    controlPlane:
      affinity:
        nodeAffinity: false
      kind: Deployment
      config:
        etcd:
          enabled: false
        scheduler:
          enabled: false
        controllerManager:
          enabled: false
        apiServer:
         staticEndpoint:
           url: "https://kubernetes.default:443"
           insecureSkipVerify: true
           auth:
             type: bearer
```
Data reporting example
https://onenr.io/0e1wZMANaQ6

Reply:
Added!